dvc_azure/__init__.py

```diff
@@ -85,87 +85,33 @@ def _get_kwargs_from_urls(urlpath: str) -> Dict[str, Any]:
-        )
-        login_info["account_name"] = config.get(
-            "account_name", _az_config().get("storage", "account", None)
-        )
-        login_info["account_key"] = config.get(
-            "account_key", _az_config().get("storage", "key", None)
-        )
-        login_info["sas_token"] = config.get(
-            "sas_token", _az_config().get("storage", "sas_token", None)
-        )
-        login_info["tenant_id"] = config.get("tenant_id")
-        login_info["client_id"] = config.get("client_id")
-        login_info["client_secret"] = config.get("client_secret")
-
-        if not (login_info["account_name"] or login_info["connection_string"]):
-            raise AzureAuthError(
-                "Authentication to Azure Blob Storage requires either "
-                "account_name or connection_string."
-            )
-
-        secondaries = (
-            "connection_string",
-            "account_key",
-            "sas_token",
-            "tenant_id",
-            "client_id",
-            "client_secret",
-        )
-        any_secondary = any(login_info[name] for name in secondaries)
-        if (
-            login_info["account_name"]
-            and not any_secondary
-            and not config.get("allow_anonymous_login", False)
-        ):
-            login_info["credential"] = DefaultAzureCredential(
-                exclude_interactive_browser_credential=False,
-                exclude_environment_credential=config.get(
-                    "exclude_environment_credential", False
-                ),
-                exclude_visual_studio_code_credential=config.get(
-                    "exclude_visual_studio_code_credential", False
-                ),
-                exclude_shared_token_cache_credential=config.get(
-                    "exclude_shared_token_cache_credential", False
-                ),
-                exclude_managed_identity_credential=config.get(
-                    "exclude_managed_identity_credential", False
-                ),
-            )
-
-        for login_method, required_keys in [  # noqa
-            ("connection string", ["connection_string"]),
-            (
-                "AD service principal",
-                ["tenant_id", "client_id", "client_secret"],
-            ),
-            ("account key", ["account_name", "account_key"]),
-            ("SAS token", ["account_name", "sas_token"]),
-            (
-                f"default credentials ({_DEFAULT_CREDS_STEPS})",
-                ["account_name", "credential"],
+        defaults = {
+            "version_aware": None,
+            "connection_string": _az_config().get(
+                "storage", "connection_string", None
             ),
-            ("anonymous login", ["account_name"]),
-        ]:
-            if all(login_info.get(key) is not None for key in required_keys):
-                break
-        else:
-            login_method = None
```

Comment:
This was super confusing and doesn't really work anyway, because it doesn't even take into account AZURE_* env vars (e.g. account name). So just pass it down to adlfs and don't bother with this. IIRC this is from very early fsspec migration days that just accumulated into this monster.

Reply:
> because it doesn't even take into account AZURE_* env vars (e.g. account name).

it does, it take it in the  `_az_config().get("storage", "account", None)`

> So just pass it down to adlfs and don't bother with this. IIRC this is from very early fsspec migration days that just accumulated into this monster.

we can blame it for the reason of this "monster" to better understand it I guess

Reply:
Good point about az_config, I completely forgot that it does take env into account. Another reason to move this to adlfs.